docusaurus-search-local/src/client/theme/SearchBar/SearchBar.tsx

```diff
@@ -181,7 +180,11 @@ export default function SearchBar({
       }
       const mark = new Mark(root);
       mark.unmark();
-      mark.mark(keywords);
+      if (keywords.length !== 0) {
+        mark.mark(keywords);
+      }
+      setInputValue(keywords.join(" "));
+      search?.autocomplete.setVal(keywords.join(" "));
```

Comment:
Does these two lines are required? Haven't they been set  in `onClearSearch`?

Reply:
Yes because you might have typed something but not selected a match, in which case the history doesn't change. I can add a comment!

Reply:
This is to apply search terms when you reload the page, or got to a link which has `?_highlight` in it. It makes it possible to clear the marking.

Reply:
I noticed there is a bug here though, search is always null. Looking into that.

Reply:
Got it.

Reply:
> I noticed there is a bug here though, search is always null. Looking into that.

Um, the `search` is asynchronous so it will probably always be null.

Reply:
The idea was that if search is assigned then we need to clear it, otherwise it doesn't matter. But even if search is assigned and then this runs search is null so something else is not working as I think it is :) It works for the click event further down, not sure what's different.

Reply:
Made it a proper ref now so it should work for the lifetime of the component.